.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,174 @@
+import { ApiItem, ApiTypeResult, ExtensionApiMap } from '../types';
+import * as babel from '@babel/core';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { singularize } from '../utils/string';
+import { loadExtensionApis } from './api-loader';
+import os from 'os';
+import { ParallelController } from '../utils/parallel';
+
+const CPU_CORES = os.cpus().length;
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+export const getApiType = (
+  apiCategory: string,
+  apiName: string
+): ApiTypeResult => {
```

Comment:
Would `api` or `namespace` and `propertyName` be better names? If so we could make that change across the code.